src/cluster/predefine.cpp

```diff
@@ -57,7 +57,7 @@ struct dynamic_converter<predefine_cfg_t> {
         const dynamic_t& nodes = source.as_object().at("nodes", dynamic_t::object_t());
 
         if(nodes.as_object().empty()) {
-            throw cocaine::error_t("at least one node should be specified for predefined cluster");
+            throw cocaine::error_t("no nodes have been specified");
```

Comment:
The `cocaine::error_t` is already a system_error. There are ~50 throwing `error_t` in a codebase, mostly in constructors, so EINVAL fits perfectly for these cases.